In [17]:
import spacy
import csv
import pandas as pd

Isolating work just in the Fine Arts department

In [8]:
with open('cmoa.csv', 'rt', encoding = 'utf8') as inp, open('cmoa_fine_art.csv', 'wt', encoding = 'utf8') as out:
    writer = csv.writer(out)
    for row in csv.reader(inp):
        if row[10] != "Fine Arts":
            writer.writerow(row)

Mady silly mistake of using the wrong index, so I'll try to isolate Fine Arts again

In [9]:
with open('cmoa.csv', 'rt', encoding = 'utf8') as inp, open('cmoa_fine_art.csv', 'wt', encoding = 'utf8') as out:
    writer = csv.writer(out)
    for row in csv.reader(inp):
        if row[9] != "Fine Arts":
            writer.writerow(row)

Still having issues, realized operation was incorrect

In [20]:
# out = open('cmoa_fine_art.csv', 'wt', encoding = 'utf8')
# writer = csv.writer(out)
# inp = open('cmoa.csv', 'rt', encoding = 'utf8')
# row = row in csv.reader(inp)
# writer.writerow(row)

with open('cmoa.csv', 'rt', encoding = 'utf8') as inp, open('cmoa_fine_art.csv', 'wt', encoding = 'utf8') as out:
#     heading = next(inp)
    writer = csv.writer(out)
#     writer.writerow(heading)
    for row in csv.reader(inp):
        if row[9] == "Fine Arts":
            writer.writerow(row)

At a glance, it seems like each row has an arbitrary row of whitespace separating each of the rows that have information, so I'll attempt to remove those

In [27]:
df = pd.read_csv('cmoa_fine_art.csv')
df.to_csv('cmoa_fine_art1.csv', index = False)

I also need to remove any "Notes" since they may cause issues in the csv file format.

In [29]:
df = pd.read_csv('cmoa_fine_art1.csv', error_bad_lines = False).dropna()
df.to_csv('cmoa_fine_art2.csv', index = False)

C:\Users\minni\AppData\Local\Temp\ipykernel_33888\1268749144.py:1: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  df = pd.read_csv('cmoa_fine_art1.csv', error_bad_lines = False).dropna()


In [28]:
with open('cmoa_fine_art1.csv', 'rt', encoding = 'utf8') as inp, open('cmoa_fine_art2.csv', 'wt', encoding = 'utf8') as out:
    writer = csv.writer(out)
    for row in csv.reader(inp):
        if row[0] != "Notes:" and row[0][0] != '[':
            writer.writerow(row)

In [23]:
# with open('cmoa_fine_art.csv', 'rt', encoding = 'utf8') as inp, open('cmoa_fine_art_1.csv', 'wt', encoding = 'utf8') as out:
#     writer = csv.writer(out)
#     for row in csv.reader(inp):
#         if row != "\n":
#             writer.writerow(row)

After attempting to also copy over the header of the csv file, I realized it would be quicker to simploy copy and paste it manually

Let's also remove all the rows that do not include a nationality, which is at index 24. Alternatively, we can reference that information with the "nationality" key word

In [30]:
filter = df["nationality"] != ""
df_clean = df[filter]

In [31]:
df.to_csv('cmoa_fine_art3.csv', index = False)

Let's compare the lengths of the csv files to see if any changes were made

In [35]:
original = pd.read_csv('cmoa.csv')
results0 = pd.read_csv('cmoa_fine_art.csv')
results1 = pd.read_csv('cmoa_fine_art1.csv')
results2 = pd.read_csv('cmoa_fine_art2.csv')
results3 = pd.read_csv('cmoa_fine_art3.csv')

In [36]:
print(len(original))
print(len(results0))
print(len(results1))
print(len(results2))
print(len(results3))

28269
8972
8972
1177
1177


While some of the cleaning may have been done incorrectly or may have been repetative, the resulting 1177 rows is now much more usable than the original 28269 for this project-- 4.16% of the number of pieces we began with!

The largest change was when I reduced the scope from the entire collection to the Fine Arts department. This superficially showcases the proportion of work in the CMOA that is considered as Fine Art (although what can be defined as such is an entirely separate discussion). Interestingly enough, in this dataset, Contemporary Art is labelled as a separate department. I can therefore infer that the work in the Fine Arts department is somewhat historical.

In [43]:
df = pd.read_csv("cmoa_fine_art3.csv")
df.loc[:, "nationality"]

0                Japanese|Japanese
1                Japanese|Japanese
2                Japanese|Japanese
3                   French|Italian
4       Japanese|Japanese|Japanese
                   ...            
1172             American|American
1173             American|American
1174             American|American
1175             Japanese|Japanese
1176             Japanese|Japanese
Name: nationality, Length: 1177, dtype: object

Below, I'm printing out specific information to get a glance at how usable the information is in order to narrow down what I include in my hosted feature layer

In [45]:
df.loc[:, "death_place"]

0                    |
1                    |
2                    |
3       |Paris, France
4                   ||
             ...      
1172                 |
1173                 |
1174                 |
1175                 |
1176                 |
Name: death_place, Length: 1177, dtype: object

I'm going to try to input the nationality data into ArcGIS to see if it can recognize the phrasing as a location. If not, I will need to continue to process the data to convert the nationality to a location. If all ese fails, I may need to use alternative methods to visualize this dataset.

In [46]:
df.loc[:, "physical_location"]

0       Not on View
1       Not on View
2       Not on View
3       Not on View
4       Not on View
           ...     
1172    Not on View
1173    Not on View
1174    Not on View
1175    Not on View
1176    Not on View
Name: physical_location, Length: 1177, dtype: object

In [47]:
df.loc[:, "classification"]

0       prints
1       prints
2       prints
3       prints
4        books
         ...  
1172    prints
1173    prints
1174    prints
1175    prints
1176    prints
Name: classification, Length: 1177, dtype: object

In [48]:
df.loc[:, "party_type"]

0              Person|Person
1              Person|Person
2              Person|Person
3              Person|Person
4       Person|Person|Person
                ...         
1172           Person|Person
1173           Person|Person
1174           Person|Person
1175           Person|Person
1176           Person|Person
Name: party_type, Length: 1177, dtype: object

In [49]:
df.loc[:, "artist_id"]

0       cmoa:parties/e5115fa9-4d3b-4ce1-b4ab-665937b0f...
1       cmoa:parties/e5115fa9-4d3b-4ce1-b4ab-665937b0f...
2       cmoa:parties/e5115fa9-4d3b-4ce1-b4ab-665937b0f...
3       cmoa:parties/eccddfd9-40ae-4159-bfa4-8cb397ff8...
4       cmoa:parties/3f8d5656-1fd7-4a8e-91ff-6e8987877...
                              ...                        
1172    cmoa:parties/ba8c351a-1864-4867-bb31-c85ef426b...
1173    cmoa:parties/ba8c351a-1864-4867-bb31-c85ef426b...
1174    cmoa:parties/ba8c351a-1864-4867-bb31-c85ef426b...
1175    cmoa:parties/4920283a-25c2-49b6-97df-2a8e6e4d7...
1176    cmoa:parties/4920283a-25c2-49b6-97df-2a8e6e4d7...
Name: artist_id, Length: 1177, dtype: object

In [50]:
df.loc[:, "role"]

0           artist|publisher
1           artist|publisher
2           artist|publisher
3                     after|
4       publisher|publisher|
                ...         
1172                       |
1173                       |
1174                       |
1175                       |
1176    Printmaker|publisher
Name: role, Length: 1177, dtype: object

In [51]:
df.loc[:, "provenance_text"]

0       Bequest of Dr. James B. Austin, Pittsburgh, PA...
1       Bequest of Dr. James B. Austin, Pittsburgh, PA...
2       Bequest of Dr. James B. Austin, Pittsburgh, PA...
3       Dukes of Marlborough and Northumberland, Syon ...
4       Bequest of Dr. James B. Austin, Pittsburgh, PA...
                              ...                        
1172    Earl Purdy [1892-1971]. David L. Purdy, Pittsb...
1173    Earl Purdy [1892-1971]. David L. Purdy, Pittsb...
1174    Earl Purdy [1892-1971]. David L. Purdy, Pittsb...
1175                       Charles Husted, Pittsburgh, PA
1176                       Charles Husted, Pittsburgh, PA
Name: provenance_text, Length: 1177, dtype: object

In [52]:
df.loc[:, "creation_date_earliest"]

0       1929-05-18
1       1929-07-30
2       1929-01-01
3       1544-12-22
4       1715-01-01
           ...    
1172    1930-01-01
1173    1952-01-01
1174    1952-01-01
1175    1940-01-01
1176    1941-01-01
Name: creation_date_earliest, Length: 1177, dtype: object

In [53]:
df.loc[:, "creation_date"]

0          5/18/1929
1          7/30/1929
2               1929
3            c. 1550
4               1715
            ...     
1172            1930
1173    c. 1926-1940
1174    c. 1926-1940
1175            1940
1176            1941
Name: creation_date, Length: 1177, dtype: object

While ArcGIS was able to recognize more locations this time around, it still struggled to handle multiple nationalities, and misinterpreted many as incorrect locations. This means I need to either continue cleaning the naitonalities data, or use an alternative data point such as the birth location.

Let's do a quick test on ArcGIS using the birth location instead

Note: ArcGIS didn't allow me to use the same file for two different hosted feature layers, so I had to maek a copy of the csv

ArcGIS handleded the birth_place information much better, and only struggled to match up Tokyo. Since there weren't many unmatched plots, I went ahead and matched those manually. After this process, 221 of the 1176 works remained unmatched. In my cleaning process, since I didn't focus on birth_place, I didn't remove the values that had null or blank fiends for the birth_place. The remainig 221 works did not have birth_place information, so for the sake of this assignment they will be left out.